Deploy/deploy.sh

```diff
@@ -130,30 +139,82 @@ pass="${cmd[5]}"
 unset cmd[4] cmd[5]
 "${cmd[@]}" --password-stdin <<< "$pass"
 
-# Create a generated version of "docker-compose.yml"
+# Build API and worker
 
-rm -rf generated
-mkdir -p generated
-cp ../docker-compose.yml generated
-cd generated
-sleep 5
-sed -i '' 's%context: ./api%context: ../../api%' docker-compose.yml
-sed -i '' 's%context: ./worker%context: ../../worker%' docker-compose.yml
-sed -i '' "s%AB2D_DB_HOST=db%AB2D_DB_HOST=$DB_ENDPOINT%" docker-compose.yml
-sed -i '' "s%AB2D_DB_DATABASE=ab2d%AB2D_DB_DATABASE=$DATABASE_NAME%" docker-compose.yml
-sed -i '' "s%AB2D_DB_USER=ab2d%AB2D_DB_USER=$DATABASE_USER%" docker-compose.yml
-sed -i '' "s%AB2D_DB_PASSWORD=ab2d%AB2D_DB_PASSWORD=$DATABASE_PASSWORD%" docker-compose.yml
-sleep 5
-cd ../..
+cd "${START_DIR}"
+cd ..
 make docker-build
 sleep 5
-cd Deploy/generated
-docker-compose build
+
+# Create API and worker Dockerfiles for ECS
+
+cd "${START_DIR}"
+rm -rf generated
+mkdir -p generated/api
+mkdir -p generated/worker
+cp ../docker-compose.yml generated
+cp ./yaml/config.yml generated
+cp ../api/Dockerfile generated/api/Dockerfile.original
+cp -r ../api/target generated/api
+cp ../worker/Dockerfile generated/worker/Dockerfile.original
+cp -r ../worker/target generated/worker
 sleep 5
-docker tag generated_api:latest 114601554524.dkr.ecr.us-east-1.amazonaws.com/ab2d_api:latest
-docker push 114601554524.dkr.ecr.us-east-1.amazonaws.com/ab2d_api:latest
-docker tag generated_worker:latest 114601554524.dkr.ecr.us-east-1.amazonaws.com/ab2d_worker:latest
-docker push 114601554524.dkr.ecr.us-east-1.amazonaws.com/ab2d_worker:latest
+cd python3
+./create-dockerfilles-for-ecs.py
+
+# Build API docker image
+
+cd "${START_DIR}"
+cd generated/api
+docker build \
```

Comment:
@lhanekam My original intention was to build a _single_ image that could be used across the environments and pass the environment variables to _containers_ instead of "baking" them into images. Let's discuss to see if there is anything preventing that.

Reply:
OK, no problem. I will modify this process based on your comment.